src/controller/CHIPDeviceController.h

```diff
@@ -393,10 +427,17 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public SessionEst
     void OnSessionEstablishmentError(CHIP_ERROR error) override;
     void OnSessionEstablished() override;
 
+    //////////// DeviceAddressUpdateDelegate Implementation ///////////////
+#if CHIP_DEVICE_CONFIG_ENABLE_MDNS
```

Comment:
should we have this `#ifdef-ed` out? I wonder if we should have methods implemented but if not using mdns just not called.

What is the logic behind mdns or not since chip requires discovery in the spec? Is this mDNS vs thread SRP?

Reply:
Honestly, I don't know why MDNS is optional in the first place, but none of the MDNS stuff is available if that config flag isn't turned on. I didn't add the flag, I'm just rolling with it.

Reply:
mDNS is not available on Thread devices.